app-creator/src/widgets/tool-bar/tool-bar.ts

```diff
@@ -35,15 +42,114 @@ export class ToolBar extends LitElement {
       padding: 0;
       font-size: 1rem;
     }
+
+    #dialog {
+      padding: var(--tight);
+      border-radius: var(--tight);
+      width: 40%;
+      max-height: 600px;
+    }
+
+    #json-string-container {
+      margin: 16px;
+      overflow-y: scroll;
+      overflow-x: scroll;
+      padding: 16px;
+      background-color: var(--background-color);
+      max-height: 400px;
+    }
+
+    #json-snippet {
+      font-family: monospace;
+    }
+
+    #copy-button {
+      background-color: var(--accent-color);
+    }
+
+    #cancel-button {
+      color: var(--accent-color);
+    }
+
+    paper-button {
+      margin-right: var(--tight);
+    }
   `;
 
+  openDialog() {
+    const dialog = this.shadowRoot?.getElementById('dialog');
+    const jsonSnippetContainer = this.shadowRoot?.getElementById(
+      'json-snippet'
+    );
+
+    if (dialog == null || jsonSnippetContainer == null) {
+      return;
+    }
+    jsonSnippetContainer.innerHTML = this.getTemplateString();
+
+    (dialog as PaperDialogElement).open();
+  }
+
+  getTemplateString() {
+    const template = store.getState().template;
+    // Remove widget reference from JSON.
+    for (const key in template) {
+      if (typeof template[key] === 'object' && !Array.isArray(template[key])) {
+        delete template[key].widgetRef;
+      }
+    }
+    return JSON.stringify(template, undefined, 3);
+  }
+
+  copy() {
+    const copyText = this.shadowRoot?.getElementById('json-snippet');
+    if (copyText == null) {
+      return;
+    }
+    const textArea = document.createElement('textarea');
+    textArea.value = copyText.innerText;
+    document.body.appendChild(textArea);
+    textArea.select();
+    document.execCommand('Copy');
+    textArea.remove();
```

Comment:
Should be fine to keep the selection there, since it acts as a nice indicator that the text has been copied?

Reply:
This method creates a textarea element in memory and never really renders it, so the selection wouldn't show.